release.sh

```diff
@@ -3,8 +3,33 @@
 DIR=$(cd $(dirname ${BASH_SOURCE[0]}) && pwd)
 cd $DIR
 
-set -e
+# set -e # do we need this? 
 
+git remote update
+
+remote=$(git rev-parse "@{u}") # '@{u}' gets the current upstream branch
+local=$(git rev-parse @) # '@' gets the current local branch 
+
+# check if local commit syncs with remote 
+if [ "$remote" != "$local" ]
+    then
+        echo "Error: local commit does not match remote. Exiting release script." 
+        exit 1
+fi 
+
+# remove any excess brackets, 'origin' branches and space/tab characters then sort the tags
+remoteTags=$(git ls-remote --tags origin | grep -v '{}' | sort | tr -d [[:blank:]])
+localTags=$(git show-ref --tags | grep -v '{}' | grep -v 'origin'| sort | tr -d [[:blank:]])
+
+# check if sorted local tags match remote tags
+if ! diff -s --left-column <(printf "%s" "$remoteTags") <(printf "%s" "$localTags") >/dev/null 2>&1
+    then 
+        echo "Error: local tags do not match remote. Exiting release script."
+        exit 1
+fi
+
+# cut the next Flux release
 version=$(go run github.com/influxdata/changelog nextver)
-git tag -s -m "Release $version" $version
+git tag -s -m "Release $version" "$version"
 git push origin "$version"
```

Comment:
One way to test this is to change these lines to `echo ....` and then run the script knowing that it will not do anything except echo these last commands. 

Also you can clone the repo into a temp directory, then clone from that directory into yet another directory. Then the first becomes the upstream of the second. 

Something like:

```
/tmp/flux-test/
      origin/ git clone from github
      working/ git clone from ./origin
```

Then you can add commits and tags to the repo in `origin` and then run the script in `working` to see if it works. All of this can remain local without having to touch github.

Reply:
Yeah I've been commenting out the final lines and printing the comment on like 31 to test the rest of script for now. That just feels like a potentially brittle way of ensuring the code works. Jonathan offered to help me test it on a cloned repo sometime this afternoon.

Reply:
Succesfully tested! Thanks for the suggestion @nathanielc!